### PR TITLE
Refactor(eos_designs)!: Only render IGMP snooping querier version and address when enabled

### PR DESCRIPTION
**Pull request by MaheshGSLAB**
## Change Summary

Render ip igmp snooping for version and address only when igmp snooping querier is enabled.

## Related Issue(s)

Fixes https://github.com/aristanetworks/avd-internal/issues/164

## Component(s) name

`arista.avd.eos_designs`

## Proposed changes
Render ip igmp snooping for version and address only when igmp snooping querier is enabled.

## How to test
Run molecule

### User Checklist

- N/A

### Repository Checklist

- [x] My code has been rebased from devel before I start
- [x] I have read the [**CONTRIBUTING**](https://avd.sh/en/latest/docs/contribution/overview.html) document.
- [x] My change requires a change to the documentation and documentation have been updated accordingly.
- [x] I have updated [molecule CI](https://github.com/aristanetworks/avd/tree/devel/ansible_collections/arista/avd/molecule) testing accordingly. (check the box if not applicable)

### CODE REVIEW

**Review by Shivani-gslab — COMMENTED** [review]
Its a breaking change, so please check if we need to add this in porting-guide and release notes.
Everything else looks good to me!

File: ansible_collections/arista/avd/docs/porting-guides/5.x.x.md [hunk 1]
```diff
@@ -442,6 +442,23 @@ To retain the previous behaviour, set the value to `false`:
 +shutdown_interfaces_towards_undeployed_peers: false
 ```
 
+### Render `ip igmp snooping vlan` command for version and address only when IGMP snooping querier is enabled
```
Comment (ClausHolbechArista) [inline ansible_collections/arista/avd/docs/porting-guides/5.x.x.md]:
```suggestion
### Only render IGMP snooping querier version and address when enabled
```

File: ansible_collections/arista/avd/docs/porting-guides/5.x.x.md [hunk 2]
```diff
@@ -442,6 +442,23 @@ To retain the previous behaviour, set the value to `false`:
...
+
+In AVD 4.0.0, the `ip igmp snooping vlan` command for version and address was rendered even when the IGMP snooping querier was not enabled.
+Starting from AVD 5.0.0, the `ip igmp snooping vlan` command for version and address is rendered only when the IGMP snooping querier is enabled.
+
+To retain the old configuration use custom structure configuration.
```
Comment (ClausHolbechArista) [inline ansible_collections/arista/avd/docs/porting-guides/5.x.x.md]:
```suggestion
To retain the old configuration use custom structured configuration.
```

File: ansible_collections/arista/avd/docs/release-notes/5.x.x.md [hunk 3]
```diff
@@ -156,6 +156,12 @@ Starting AVD 5.0.0, the default MTU for Dps1, WAN HA and WAN p2p uplink interfac
 
 See the [porting guide](../porting-guides/5.x.x.md#default-mtu-value-changed-from-9214-to-9194-for-dps1-wan-ha-and-wan-p2p-uplink-interfaces) for details.
 
+#### Render `ip igmp snooping vlan` command for version and address only when IGMP snooping querier is enabled
```
Comment (ClausHolbechArista) [inline ansible_collections/arista/avd/docs/release-notes/5.x.x.md]:
```suggestion
#### Only render IGMP snooping querier version and address when enabled
```

**Review by Shivani-gslab — APPROVED** [review]
LGTM

**Review by ClausHolbechArista — APPROVED** [review]
LGTM

**Pull request merged**